tests/integration/tier0/check-internet-connection/test_internet_connection.py

```diff
@@ -0,0 +1,61 @@
+from envparse import env
+
+
+def _modify_dnsmasq():
+    with open("/etc/dnsmasq.conf", "a") as f:
+        # Everything else is resolved to localhost
+        f.write("address=/#/127.0.0.1")
+
+    with open("/etc/resolv.conf", "w") as f:
+        f.write("nameserver 127.0.0.1")
+
+
+def _remove_changes_from_dnsmasq():
+    """Delete only the last line that was appended to both files."""
```

Comment:
Correct me if I'm wrong but for me it seems that since you are opening the `resolv.conf` in `w` mode you are overwriting it's content no? So removing the last line in this case will not restore the file?

Reply:
So, I have changed it a bit after @abadger comments below. Decided to save the original state of the file, modify it, and then restore after the conversion. 

Seems simpler this way, and we don't have to deal with impartial rewrites too.